library/kani_macros/src/contract.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Utility functions for macro expansion of function contract clauses.
+use proc_macro2::TokenStream as TokenStream2;
+use proc_macro2::{Ident, Span};
+use quote::quote;
+use syn::{Attribute, Expr, FnArg, ItemFn, Signature};
+use uuid::Uuid;
+
+/// Given a function `foo`, this function creates a closure with name
+/// `foo_<uuid>`, where <uuid> is a unique identifier.
+/// The closure has the same function arguments and the same function body
+/// as the original function `foo`.
+pub fn convert_to_closure(item: &ItemFn) -> (Ident, TokenStream2) {
+    let fn_sig = &item.sig;
+    let body = &item.block;
+    let args_with_ty = &fn_sig.inputs;
+    let rt = &fn_sig.output;
+    let name = format!("{}_{}", fn_sig.ident.clone().to_string(), Uuid::new_v4()).replace("-", "_");
+    let ident = Ident::new(name.as_str(), Span::call_site());
+    let inner_fn = quote! {
+        let #ident = |#args_with_ty| #rt #body;
+    };
+    (ident, inner_fn)
+}
+
+/// Converts all "[#kani::ensures(...)]" attributes to "#kani::postcondition(...)" statement tokens.
+pub fn extract_ensures_as_postconditions(attributes: &Vec<Attribute>) -> TokenStream2 {
+    attributes
+        .iter()
+        .filter_map(|a| {
+            let name = a.path.segments.last().unwrap().ident.to_string();
+            match name.as_str() {
+                "ensures" => {
+                    let arg = a
+                        .parse_args::<Expr>()
+                        .expect("An argument expected inside the ensures clause");
+                    Some(quote! {kani::postcondition(#arg);})
+                }
+                _ => None,
+            }
+        })
+        .collect()
+}
+
+/// Returns the list of function arguments from the function signature.
```

Comment:
Comment that you are ignoring `self`. 
Comment explaining why

Reply:
Added a comment. The function is returning a list of arguments for use in a function or method call. 
If we have a function `fn foo(self, x: i32, y:i32) -> ...`, it extracts `vec![x, y]` for use in the `foo(x, y)` method call.